### ISSUE TITLE
Support org.osgi.framework version 1.7.0

### ISSUE THREAD

**Issue by falmanna** [openhab/openhab-core#1034]
After switching to the new build system starting from the M2 release, bindings are not compatible anymore with Concierge OSGi framework. The stable version `5.1.0` of Concierge framework provides `org.osgi.framework = 1.7.0` only, and since the M2 release, all bindings requires `osgi.extender`, specifically, the `osgi.component 1.4.0<=version<=2.0.0` which requires `org.osgi.framework = 1.8.2`

**Comment by kaikreuzer** [comment]
@maggu2810 Do you maybe remember, what change made us require a framework of version 1.8.2 or higher?

**Comment by maggu2810** [comment]
Didn't we allowed to use R6 instead of sticking to R4.2 while we migrated from ESH to OH again?

**Comment by kaikreuzer** [comment]
Hm, might be, but did we have a good reason for it?
At least I wasn't so far aware that we broke Concierge compatibility, which was always a feature of ESH that we ensured to work, so it is a bit sad to lose it suddenly.
But sure, if we have reasons to require 1.8.2, that's how it is then.

**Comment by maggu2810** [comment]
At runtime we are using DS 1.4 features like the constructor injection.

You can check the requirements of the code base itself by changing the version of the OSGi Core API yourself (here: https://github.com/openhab/openhab-core/blob/fa416be/bom/compile/pom.xml#L24-L26).

The first compile time failure is:

```
[ERROR] .../openhab-core/bundles/org.openhab.core.test/src/main/java/org/eclipse/smarthome/test/internal/java/MissingServiceAnalyzer.java:[75,3186] The type org.osgi.framework.dto.BundleDTO cannot be resolved. It is indirectly referenced from required .class files
[ERROR] .../openhab-core/bundles/org.openhab.core.test/src/main/java/org/eclipse/smarthome/test/internal/java/MissingServiceAnalyzer.java:[75,3186] BundleDTO cannot be resolved to a type
```

**Comment by kaikreuzer** [comment]
@falmanna If R5 support is important to you, please follow @maggu2810's suggestion and check if there are any real blockers for it (tests aren't really an issue as we could potentially allow them to compile against a newer version).
In general, I would otherwise tend to stick to the decision that the code requires R6 (and thus drop Concierge support until there's a R6 compatible version available).

**Comment by falmanna** [comment]
Thanks, @kaikreuzer @maggu2810 

It looks like only the tests are failing, I compiled the tests against R6 and switched the rest to R5 and it compiled successfully. 

But the problem still exists when running bundles under Concierge. The bundles still shows `[BundleRequirement {osgi.extender; filter:=(&(osgi.extender=osgi.component)(version>=1.4.0)(!(version>=2.0.0)))}`

I tried downgrading `osgi.cmpn` to version 5.0.0, then I got the error 
`osgi The annotation @Activate/@Reference is disallowed for this location`
I am trying to figure out how to solve this, and I appreciate any input from you.
